wpe/src/main/cpp/browser/browser.cpp

```diff
@@ -297,3 +296,24 @@ void Browser::deleteInputMethodContent(int pageId, int offset)
         delete static_cast<InputMethodContentData*>(data);
     });
 }
+
+void Browser::updateAllSettings(int pageId, const PageSettings& settings)
+{
+    if (m_pages.find(pageId) == m_pages.end())
+        return;
+
+    struct PageSettingsData
+    {
+        Page* page;
+        PageSettings settings;
+    };
+    auto* data = new PageSettingsData { m_pages[pageId].get(), settings };
+    g_main_context_invoke_full(*m_uiProcessThreadContext, G_PRIORITY_DEFAULT, +[](gpointer data) -> gboolean {
+        auto* data_ = static_cast<PageSettingsData*>(data);
+        if (data_->page != nullptr)
+            data_->page->updateAllSettings(data_->settings);
+        return G_SOURCE_REMOVE;
+    }, data, +[](gpointer data) {
+        delete static_cast<PageSettingsData*>(data);
+    });
```

Comment:
Good point. Will fix this. Just copy pasted the logic from other functions in the file which used similar convention. Didn't spent much time thinking on this because eventual plan is to refactor whole thing with architecture redesign task